File: Java/프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int solution(int[][] sizes) {
        List<Integer> minArr = new ArrayList<>();
        List<Integer> maxArr = new ArrayList<>();
        for (int[] size : sizes) {
            int min = Math.min(size[0], size[1]);
            int max = Math.max(size[0], size[1]);
            minArr.add(min);
            maxArr.add(max);
        }
        int maxOfMin = minArr.stream().mapToInt(Integer::intValue).max().orElse(0);
        int maxOfMax = maxArr.stream().mapToInt(Integer::intValue).max().orElse(0);

        int answer = maxOfMin * maxOfMax;
        return answer;
    }
}

File: Kotlin/프로그래머스/1/86491. 최소직사각형/최소직사각형.kt
import kotlin.math.*
class Solution {
    fun solution(sizes: Array<IntArray>): Int {
        var maxA = 0
        var maxB = 0

        for (l in sizes) {
            l.sort()
            maxA = max(maxA, l[0])
            maxB = max(maxB, l[1])
        }

        return maxA * maxB
    }
}